File: core/src/main/java/com/toocol/ssh/core/file/core/FileChooser.java
package com.toocol.ssh.core.file.core;

import com.toocol.ssh.utilities.console.Console;

/**
 * @author ：JoeZane (joezane.cn@gmail.com)
 * @date: 2022/4/15 22:07
 * @version: 0.0.1
 */
public class FileChooser {

    private static final Console CONSOLE = Console.get();

    /**
     * multiple file paths separate by ','
     *
     * @return file paths
     */
    public String showOpenDialog() {
        String files = CONSOLE.chooseFiles();
        return files == null ? null : files.substring(0, files.length() - 1);
    }

}

File: core/src/main/java/com/toocol/termio/core/file/core/FileChooser.kt
package com.toocol.termio.core.file.core

import com.toocol.termio.utilities.console.Console

/**
 * @author ：JoeZane (joezane.cn@gmail.com)
 * @date: 2022/4/15 22:07
 * @version: 0.0.1
 */
class FileChooser {
    /**
     * multiple file paths separate by ','
     *
     * @return file paths
     */
    fun showOpenDialog(): String? {
        val files = console.chooseFiles()
        return files?.substring(0, files.length - 1)
    }

    companion object {
        private val console = Console.get()
    }
}